vindemitor/core/drm/widevine.py

```diff
@@ -203,89 +203,121 @@ def decrypt(self, path: Path) -> None:
         """
         if not self.content_keys:
             raise ValueError("Cannot decrypt a Track without any Content Keys...")
 
-        if not binaries.ShakaPackager:
-            raise EnvironmentError("Shaka Packager executable not found but is required.")
+        if config.general.decryptor == "shaka":
+            if not binaries.ShakaPackager:
+                raise EnvironmentError("Shaka Packager executable not found but is required.")
+        elif config.general.decryptor == "mp4decrypt":
+            if not binaries.Mp4Decrypt:
+                raise EnvironmentError("mp4decrypt executable not found but is required.")
         if not path or not path.exists():
             raise ValueError("Tried to decrypt a file that does not exist.")
 
         output_path = path.with_stem(f"{path.stem}_decrypted")
         config.directories.temp.mkdir(parents=True, exist_ok=True)
 
-        try:
-            arguments = [
-                f"input={path},stream=0,output={output_path},output_format=MP4",
-                "--enable_raw_key_decryption",
-                "--keys",
-                ",".join(
-                    [
-                        *[
-                            "label={}:key_id={}:key={}".format(i, kid.hex, key.lower())
-                            for i, (kid, key) in enumerate(self.content_keys.items())
-                        ],
-                        *[
-                            # some services use a blank KID on the file, but real KID for license server
-                            "label={}:key_id={}:key={}".format(i, "00" * 16, key.lower())
-                            for i, (kid, key) in enumerate(self.content_keys.items(), len(self.content_keys))
-                        ],
-                    ]
-                ),
-                "--temp_dir",
-                config.directories.temp,
-            ]
-
-            p = subprocess.Popen(
-                [binaries.ShakaPackager, *arguments],
-                stdout=subprocess.DEVNULL,
-                stderr=subprocess.PIPE,
-                universal_newlines=True,
-            )
+        if config.general.decryptor == "mp4decrypt":
+            try:
+                key_arg = []
+                for kid, key in self.content_keys.items():
+                    kid_hex = kid.hex if hasattr(kid, "hex") else str(kid).replace("-", "")
+                    key_hex = key if isinstance(key, str) else key.hex()
+                    key_arg.extend(["--key", f"{kid_hex}:{key_hex}"])
+
+                cmd = [
+                    str(binaries.Mp4Decrypt),
+                    *key_arg,
+                    str(path), # Input
+                    str(output_path) # Output
+                ]
+
+                subprocess.run(cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+            except subprocess.CalledProcessError as e:
+                if e.returncode == 0xC000013A:  # STATUS_CONTROL_C_EXIT
```

Comment:
Create a constant for the return code, assign the constant based on OS. Current code is Windows specific.